datasync/chaindatafetcher/chaindata_fetcher.go

```diff
@@ -382,14 +413,19 @@ func (f *ChainDataFetcher) handleRequest() {
 			return
 		case ev := <-f.chainCh:
 			numChainEventGauge.Update(int64(len(f.chainCh)))
+			var err error
 			switch f.config.Mode {
 			case ModeKAS:
-				f.handleRequestByType(cfTypes.RequestTypeAll, true, ev)
+				err = f.handleRequestByType(cfTypes.RequestTypeAll, true, ev)
 			case ModeKafka:
-				f.handleRequestByType(cfTypes.RequestTypeGroupAll, true, ev)
+				err = f.handleRequestByType(cfTypes.RequestTypeGroupAll, true, ev)
 			default:
 				logger.Error("the chaindatafetcher mode is not supported", "mode", f.config.Mode, "blockNumber", ev.Block.NumberU64())
 			}
+
+			if err != nil && err == errMaxRetryExceeded {
+				f.forceStop()
```

Comment:
Shouldn't we return here to terminate `handleRequest`?
```suggestion
				f.forceStop()
				return
```